### PR TITLE
Stepwise mutation model

### PR DESCRIPTION
**Pull request by thoree**
The stepwise model simillar to Simonsson and Mostad (FSI: Gentics 2015) has been implemented and documented.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Install from GitHub as follows:
 
 ``` r
 # install.packages("devtools")
-devtools::install_github("magnusdv/pedmut")
+devtools::install_github("thoree/pedmut")
```
Comment (magnusdv) [inline README.md]: Perhaps best to revert this change.

File: README.Rmd [hunk 2]
```diff
@@ -59,12 +59,23 @@ The mutations models currently implemented in `pedmut` are:
 * "proportional" : Mutation probabilities are proportional to the target allele frequencies. Parameters: `rate`, `afreq`.
 * "random" : This produces a matrix of random numbers, each row normalised to have sum 1. Parameters: `seed`.
 * "custom" : Allows any valid mutation matrix to be provided by the user. Parameters: `matrix`.
+* "trivial" : Diagonal mutation matrix with 1 on the diagonal.
+* "stepwise" : Mutation rates depend on whether transitions are within the same group or not, i.e., between integer alleles (like '17') and microvariants (like '17.1'). Mutations also depend on the size of the mutation as modelled by the parameter 'range', the relative probability of mutating n+1 steps versus mutating n steps.
```
[the remark this answers is not in the document]
Comment (magnusdv) [reply README.Rmd]: Better now, thanks!

File: R/mutationMatrix.R [hunk 3]
```diff
@@ -105,17 +105,19 @@ mutationMatrix = function(model = c("custom", "equal", "proportional",
...
   if(model %in% c("stepwise")) {
     if(is.null(rate2))
-      stop("`rate2` cannot be NULL with this model")
-    if(rate2 < 0 | rate2 > 1)
-      stop("`rate2` must be in [0,1]: ", rate2)
+      stop("`rate2` cannot be NULL with the `stepwise` model")
+    if(!is_number(rate2, minimum = 0))
+      stop2("`rate2` must be a nonnegative number: ", rate2)
+    if(!is_number(rate + rate2, maximum = 1))
+      stop("The total mutation rate `rate + rate2` must be in [0,1]: ", rate + rate2)
```
Comment (magnusdv) [inline R/mutationMatrix.R]: Aha - nice.
But I guess at this point both `rate` and  `rate2` are confirmed scalars, so `if (rate + rate2 > 1)` would suffice.

Comment (thoree) [reply R/mutationMatrix.R]: You are right as always!

**Comment by magnusdv** [comment]
Does devtools::check() run cleanly now?

**Comment by thoree** [comment]
yes,  devtools::check() runs cleanly based on the version I pushed to
https://github.com/thoree/pedmut:

R CMD check results
0 errors | 0 warnings | 0 notes

Please let me know if there's more I need to do.
